src/NHibernate/SqlCommand/SqlInsertBuilder.cs

```diff
@@ -66,6 +66,8 @@ public virtual SqlInsertBuilder AddColumn(string columnName, IType propertyType)
 		/// <param name="val">The value to set for the column.</param>
 		/// <param name="literalType">The NHibernateType to use to convert the value to a sql string.</param>
 		/// <returns>The SqlInsertBuilder.</returns>
+		// Since v5.6
+		[Obsolete("This method is unsafe and has no more usages. Use the overload with a property type and use a parameterized query.")]
 		public SqlInsertBuilder AddColumn(string columnName, object val, ILiteralType literalType)
 		{
 			return AddColumn(columnName, literalType.ObjectToSQLString(val, Dialect));
```

Comment:
As noted by @gliljas in #3516, these methods implementations are unsafe and may lead to SQL injections or invalid SQL. Since these `AddColumn` overloads have no usage, better just obsolete them. Adding a value in a SQL query as a string fragment is anyway a bad practice. Parameterization should be used instead.

Reply:
I'm curious why not remove it?

Reply:
Thanks!